### PR TITLE
Add jest-axe to test basic a11y

### PR DESCRIPTION
**Pull request by Ladsgroup**
Bug: T265299

### CODE REVIEW

**Comment by Ladsgroup** [comment]
Rebasing this on master, it should pass now.

**Comment by micgro42** [comment]
Fixed the linting issues to showcase that it can actually get green. For the real implementation, we probably want it in a different place, i.e. in its own test job and not mixed into the unit tests.

**Review by bereket-WMDE — APPROVED** [review]

**Comment by Ladsgroup** [comment]
The test is now failing for an actual a11y issue: https://phabricator.wikimedia.org/T265245#6637658

File: tests/unit/components/QueryBuilder.spec.ts [hunk 1]
```diff
@@ -111,4 +111,30 @@ describe( 'QueryBuilder.vue', () => {
 		} );
 
 	} );
+
+	it( 'should not have obvious accessibility issues', async () => {
```
Comment (micgro42) [inline tests/unit/components/QueryBuilder.spec.ts]: could we move this out of the unit tests and into its own `a11y` test directory that is run on `npm run test:a11y`?

We might want to add more test scenarios in the future.

Comment (Ladsgroup) [reply tests/unit/components/QueryBuilder.spec.ts]: Done.

**Review by micgro42 — APPROVED** [review]

**Comment by Ladsgroup** [comment]
Manual rebase ^

**Comment by Ladsgroup** [comment]
The failure is unrelated and will be fixed in #83

**Pull request merged**